src/o365/graph/commands/teams/teams-user-add.ts

```diff
@@ -0,0 +1,170 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import Utils from '../../../../Utils';
+import { GroupUser } from '../o365group/GroupUser';
+import * as request from 'request-promise-native';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  role?: string;
+  teamId: string;
+  userName: string;
+}
+
+class GraphTeamsUserAddCommand extends GraphItemsListCommand<GroupUser> {
+  public get name(): string {
+    return `${commands.TEAMS_USER_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Adds an owner or member to the specified Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.role = args.options.role;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/users/${encodeURIComponent(args.options.userName)}/id`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none'
+          }),
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.get(requestOptions);
+      })
+      .then((res: any): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:')
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        let endpoint: string = `${auth.service.resource}/v1.0/groups/${args.options.teamId}/members/$ref`;
+
+        if (args.options.role === "Owner") {
```

Comment:
Since the only difference between the URL for members and owners is the role segment, you could simplify it like:

```ts
const endpoint: string = `${auth.service.resource}/v1.0/groups/${args.options.teamId}/${args.options.role === 'Owner' ? 'owners' : 'members'}/$ref`;
```